c10/core/TensorOptions.h

```diff
@@ -337,7 +337,7 @@ struct C10_API TensorOptions {
 
   // Resolves the tensor type set specified by the current construction axes.
   DispatchKeySet key_set() const noexcept {
-    return DispatchKeySet(computeDispatchKey());
+    return DispatchKeySet(computeDispatchKey()).add(DispatchKey::BackendSelect);
```

Comment:
This can go in the global selector now that

Reply:
what do you mean under global selector?

Reply:
@ezyang: I think your comment got cut off.

My question would be: why do we add BackendSelect but not Variable?

Reply:
```
  // Some keys are ALWAYS considered for inclusion by default, so they are
  // included in the set here.  (const appears to be sufficient for
  // always_included to get inlined, constexpr not necessary)
  const DispatchKeySet always_included{DispatchKey::VariableTensorId};
```

in `DispatchKeyExtractor.h`

This is why Variable doesn't need to be added here anymore, it's handled by `always_included`

Reply:
right, so can we either unify those places or explain in a comment why they are different?

Reply:
Added a comment.